internal/utils/http/throttling-aware-client.go

```diff
@@ -18,7 +18,7 @@ func NewThrottlingAwareClient(httpClient *http.Client, requestLogger func(format
 
 	client.RetryMax = 15
 	client.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
-		return resp.StatusCode == 429, nil
+		return resp == nil || resp.StatusCode == 429, nil
```

Comment:
this is very hard to read O:)

Reply:
I know, unfortunately, it _has_ happened that it was `nil` so that check is needed... And, yes, I could split it into multiple lines, but... :innocent:

Reply:
I'm fine with checking for nil. I'd just prefer having it split in multiple lines for readability.